python/ray/serve/batching.py

```diff
@@ -228,13 +253,18 @@ async def _process_batches(self, func: Callable) -> None:
                     results = await func_future
                     self._validate_results(results, len(batch))
                     for result, future in zip(results, futures):
-                        # If the client has disconnected, the future is canceled.
-                        # We should only set result when the connection is still live.
-                        if not future.cancelled():
-                            future.set_result(result)
+                        _set_result_if_not_done(future, result)
```

Comment:
Let's extract non-gen handling into a method as well (mirroring generator)

Reply:
I moved the handling into a new method, `_assign_func_results`.